### PR TITLE
Final detection layer

### PR DESCRIPTION
**Pull request by jhung0**
Right now the model's output isn't exactly the boxes you'd be interested in, so this layer should output those boxes/labels.
I'm not sure what a good name would be for it though.

### CODE REVIEW

File: keras_rcnn/layers/detection.py [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+
+
+        return [keras.backend.expand_dims(pred_boxes, 0), keras.backend.expand_dims(pred_scores, 0)]
+
+
+    def compute_output_shape(self, input_shape):
+        return [(1, None, 4), (1, None, 3)]
+
+    def compute_mask(self, inputs, mask=None):
+        return 2 * [None]
```
Comment (0x00b1) [inline keras_rcnn/layers/detection.py]: Missing a newline

File: keras_rcnn/layers/detection.py [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+        indices = keras_rcnn.backend.non_maximum_suppression(pred_boxes, pred_scores_classes, keras.backend.shape(pred_boxes)[0], self.TEST_NMS)
+        pred_scores = keras.backend.gather(pred_scores, indices)
+        pred_boxes = keras.backend.gather(pred_boxes, indices)
+
+
+        return [keras.backend.expand_dims(pred_boxes, 0), keras.backend.expand_dims(pred_scores, 0)]
+
+
+    def compute_output_shape(self, input_shape):
+        return [(1, None, 4), (1, None, 3)]
```
Comment (0x00b1) [inline keras_rcnn/layers/detection.py]: `3`?

Comment (0x00b1) [reply keras_rcnn/layers/detection.py]: `1` should be `None` (or `input_shape[0]`)

File: keras_rcnn/layers/detection.py [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+        pred_boxes = keras.backend.reshape(pred_boxes, (-1, 4))
+
+        indices = keras_rcnn.backend.non_maximum_suppression(pred_boxes, pred_scores_classes, keras.backend.shape(pred_boxes)[0], self.TEST_NMS)
+        pred_scores = keras.backend.gather(pred_scores, indices)
+        pred_boxes = keras.backend.gather(pred_boxes, indices)
+
+
+        return [keras.backend.expand_dims(pred_boxes, 0), keras.backend.expand_dims(pred_scores, 0)]
+
+
```
Comment (0x00b1) [inline keras_rcnn/layers/detection.py]: Remove extra newline
